helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -61,21 +61,34 @@ public static RebalanceAlgorithm getInstance(
             new ReplicaActivateConstraint(), new NodeMaxPartitionLimitConstraint(),
             new ValidGroupTagConstraint(), new SamePartitionOnInstanceConstraint());
 
-    int evennessPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS, 1);
-    int movementPreference =
-        preferences.getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT, 1);
+    int evennessPreference = 1;
+    int movementPreference = 1;
+    if (preferences.containsKey(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS) && preferences
+        .containsKey(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS)) {
+      evennessPreference = preferences.get(ClusterConfig.GlobalRebalancePreferenceKey.EVENNESS);
+      movementPreference =
+          preferences.get(ClusterConfig.GlobalRebalancePreferenceKey.LESS_MOVEMENT);
+    }
+
+    boolean forceBaselineConverge = preferences
+        .getOrDefault(ClusterConfig.GlobalRebalancePreferenceKey.FORCE_BASELINE_CONVERGE, 0)
+        > 0;
+    if (forceBaselineConverge) {
+      MODEL.put(PartitionMovementConstraint.class.getSimpleName(), 0.001f);
+      MODEL.put(BaselineInfluenceConstraint.class.getSimpleName(), 10000f);
+    }
 
     List<SoftConstraint> softConstraints = ImmutableList
-        .of(new PartitionMovementConstraint(), new InstancePartitionsCountConstraint(),
-            new ResourcePartitionAntiAffinityConstraint(),
+        .of(new PartitionMovementConstraint(), new BaselineInfluenceConstraint(),
+            new InstancePartitionsCountConstraint(), new ResourcePartitionAntiAffinityConstraint(),
             new ResourceTopStateAntiAffinityConstraint(), new MaxCapacityUsageInstanceConstraint());
-    Map<SoftConstraint, Float> softConstraintsWithWeight = Maps.toMap(softConstraints, key -> {
-      String name = key.getClass().getSimpleName();
-      float weight = MODEL.get(name);
-      return name.equals(PartitionMovementConstraint.class.getSimpleName()) ?
-          movementPreference * weight : evennessPreference * weight;
-    });
+    Map<SoftConstraint, Float> softConstraintsWithWeight = new HashMap<>();
+    for (SoftConstraint softConstraint : softConstraints) {
+      float constraintWeight = MODEL.get(softConstraint.getClass().getSimpleName());
+      softConstraintsWithWeight.put(softConstraint,
```

Comment:
Comment on why we treat BaselineInfluenceConstraint as an evennessPreference instead of movementPreference please.